use-cases/virgo/hpo.py

```diff
@@ -0,0 +1,225 @@
+import argparse
+import os
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import ray
+import torch
+from ray import train, tune
+
+from itwinai.parser import ConfigParser
+
+
+def run_trial(config):
+    """
+    Execute a single trial using the given configuration (config).
+    This runs a full training pipeline - you can also specify a pipeline as a dictionary, 
+    e.g. if you only want to run certain parts without changing your config.yaml file (see below).
+
+    Args:
+    - config: Dictionary with hyperparameters (e.g., 'batch_size', 'lr').
+
+    Example to run with a manual pipeline (you would have to import these classes):
+
+    my_pipeline = Pipeline(
+        [
+            TimeSeriesDatasetSplitter(
+                train_proportion=0.9,
+                root_folder="/p/scratch/intertwin/datasets/virgo"
+            ),
+            NoiseGeneratorTrainer(
+                config=config,
+                num_epochs=4,
+                strategy=None,
+                checkpoint_path='checkpoints/checkpoint_epoch_{}.pth',
+                validation_every=20
+            )
+        ]
+    )
+
+    Passing a seed to TimeSeriesDatasetSplitter and NoiseGeneratorTrainer
+    will make runs uniform across trials
+    (reducing the variablility to the hyperparameter settings)
```

Comment:
It is a bit unclear to me what this example shows. Is this what the `run_trial()` function does or is it meant to show me how I would use this function myself? In this case, I am not sure how I would do that as the example doesn't use the `run_trial()` function itself. Consider rephrasing this to make the intent a bit more clear.